sunpy/net/base_client.py

```diff
@@ -31,15 +30,46 @@ def __init_subclass__(cls, *args, **kwargs):
         This means that Fido can use the clients internally.
         """
         super().__init_subclass__(**kwargs)
+        # Circular import
+        from sunpy.net import attr
         # We do not want to register GenericClient since its a dummy client.
         if cls.__name__ != 'GenericClient':
             cls._registry[cls] = cls._can_handle_query
+            # This can't be a good way to do it.
+            attr.Attr.update_values(cls.register_values())
+
+    def __repr__(self):
+        """
+        This enables the "pretty" printing of clients.
+        """
+        class_name = self.__class__.__name__
+        attrs = self.register_values()
+        lines = []
+        attr_name = []
+        attr_values = []
+        attr_desc = []
+        for key in attrs.keys():
+            attr_name.append(key.__name__)
+            attr_values.append(attrs[key][0][0])
+            attr_desc.append(attrs[key][0][1])
+        pad_name = max(len(elm) for elm in attr_name)
+        pad_long = max(len(elm) for elm in attr_values)
+        pad_desc = max(len(elm) for elm in attr_desc)
+        fmt = f"%-{pad_name}s | %-{pad_long}s | %-{pad_desc}s"
+        lines = [fmt % elm for elm in zip(attr_name, attr_values, attr_desc)]
+        lines.insert(0, class_name)
+        lines.insert(1, self.__doc__.partition("\n\n")[0]+"\n")
+        lines.insert(2, "Attr".ljust(pad_name + 2) + "Value".ljust(pad_long + 3) +
+                     "Description".ljust(pad_desc + 1))
+        lines.insert(3, '-' * (pad_name + 1) + '+' + '-' * (pad_long + 2) + '+' + '-' * (pad_desc + 1))
+        return "\n".join(lines)
```

Comment:
This is so ugly.

Reply:
The only other option is build an astropy table and use the `pprint` method.

Reply:
IS THERE NO OTHER CHOICE?

Reply:
:laughing: probably, but I can't think of one.

Reply:
Probably won't do that in this PR just for my sanity.